src/libraries/System.Net.Quic/tests/FunctionalTests/QuicTestBase.cs

```diff
@@ -28,6 +30,10 @@ public abstract class QuicTestBase<T>
         public X509Certificate2 ServerCertificate = System.Net.Test.Common.Configuration.Certificates.GetServerCertificate();
         public X509Certificate2 ClientCertificate = System.Net.Test.Common.Configuration.Certificates.GetClientCertificate();
 
+        public const int PassingTestTimeout = 30_000;
```

Comment:
Why don't you declare it as `TimeSpan`? You wouldn't need to add those overloads to TaskTimeoutExtensions.cs and you'd have avoided mistakes like `WaitAsync(TimeSpan.FromSeconds(PassingTestTimeout))` - seconds instead of milliseconds. Unless you really want there super long timeout.

Reply:
Some tests suites have actually both - one created from the other. I may do that for consistency.